packages/server/lib/modes/run.ts

```diff
@@ -0,0 +1,1661 @@
+  }
+
+  // user code might have modified list of allowed browsers
+  // but be defensive about it
+  const userBrowsers = _.get(config, 'resolved.browsers.value', browsers)
+
+  let specPattern = specPatternFromCli || config.specPattern
+
+  specPattern = relativeSpecPattern(projectRoot, specPattern)
+
+  const [sys, browser] = await Promise.all([
+    system.info(),
+    (async () => {
+      const browsers = await browserUtils.ensureAndGetByNameOrPath(browserName, false, userBrowsers)
+
+      await removeOldProfiles(browsers)
+
+      return browsers
+    })(),
+    trashAssets(config),
+  ])
+
+  // @ts-expect-error ctx is protected
+  const specs = project.ctx.project.specs
+
+  if (!specs.length) {
+    errors.throwErr('NO_SPECS_FOUND', projectRoot, String(specPattern))
+  }
+
+  if (browser.unsupportedVersion && browser.warning) {
+    errors.throwErr('UNSUPPORTED_BROWSER_VERSION', browser.warning)
+  }
+
+  if (browser.family === 'chromium') {
+    chromePolicyCheck.run(onWarning)
+  }
+
+  async function runAllSpecs ({ beforeSpecRun, afterSpecRun, runUrl, parallel }: { beforeSpecRun?: BeforeSpecRun, afterSpecRun?: AfterSpecRun, runUrl?: string, parallel?: boolean}) {
+    const results = await runSpecs({
+      beforeSpecRun,
+      afterSpecRun,
+      projectRoot,
+      socketId,
+      parallel,
+      onError,
+      browser,
+      project,
+      runUrl,
+      group,
+      config,
+      specs,
+      sys,
+      tag,
+      specPattern,
+      videosFolder: config.videosFolder,
+      video: config.video,
+      videoCompression: config.videoCompression,
+      videoUploadOnPasses: config.videoUploadOnPasses,
+      headed: options.headed,
+      quiet: options.quiet,
+      outputPath: options.outputPath,
+      testingType: options.testingType,
+      exit: options.exit,
+      webSecurity: options.webSecurity,
+    })
+
+    if (!options.quiet) {
+      renderSummaryTable(runUrl)(results)
+    }
+
+    return results
+  }
+
+  if (record) {
+    const { projectName } = config
+
+    return recordMode.createRunAndRecordSpecs({
+      tag,
+      key,
+      sys,
+      specs,
+      group,
+      config,
+      browser,
+      parallel,
+      ciBuildId,
+      testingType,
+      project,
+      projectId,
+      projectRoot,
+      projectName,
+      specPattern,
+      runAllSpecs,
+      onError,
+      quiet: options.quiet,
+    })
+  }
+
+  // not recording, can't be parallel
+  return runAllSpecs({
+    parallel: false,
+  })
+}
+
+export async function run (options, loading: Promise<void>) {
+  if (require('../util/electron-app').isRunningAsElectronProcess({ debug })) {
+    const app = require('electron').app
+
+    // electron >= 5.0.0 will exit the app if all browserwindows are closed,
+    // this is obviously undesirable in run mode
+    // https://github.com/cypress-io/cypress/pull/4720#issuecomment-514316695
+    app.on('window-all-closed', () => {
```

Comment:
I wonder if this is still true

Reply:
Yeah, it should be. This was an intentional change, since most Electron apps should have 0 processes left when all the windows are closed. We're the exception to their rule

Reply:
Here's the underlying Electron code that handles this, for reference: https://github.com/electron/electron/blob/d44a187d0b226800fe0cb4f7a0d2b36c871b27cd/lib/browser/init.ts#L172